packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/RecorderHostApiImpl.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.video.FileOutputOptions;
+import androidx.camera.video.PendingRecording;
+import androidx.camera.video.Recorder;
+import androidx.core.content.ContextCompat;
+
+import java.io.File;
+import java.util.Objects;
+
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.RecorderHostApi;
+
+public class RecorderHostApiImpl implements RecorderHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+  private Context context;
+
+  @VisibleForTesting public CameraXProxy cameraXProxy = new CameraXProxy();
+
+  @VisibleForTesting public PendingRecordingFlutterApiImpl pendingRecordingFlutterApi;
+
+  public RecorderHostApiImpl(
+      BinaryMessenger binaryMessenger, @NonNull InstanceManager instanceManager, Context context) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    this.context = context;
+    this.pendingRecordingFlutterApi =
+        new PendingRecordingFlutterApiImpl(binaryMessenger, instanceManager);
+  }
+
+  @Override
+  public void create(@NonNull Long instanceId, Long aspectRatio, Long bitRate) {
+    Recorder.Builder recorderBuilder = cameraXProxy.createRecorderBuilder();
+    if (aspectRatio != null) {
+      recorderBuilder.setAspectRatio(Math.toIntExact(aspectRatio));
```

Comment:
I'm open to either, but it brings up a sort of general question I had. In wrapping these CameraX classes, should we strive to expose all the functionality of the original class (even when not needed), or should we only be exposing exactly what we are using? 

I had a general impression that wrapping as closely to the original class (when easy to do so) was generally better, which was why I left it in despite not using it. But again I am fine removing if you think that is better!

Reply:
We discussed this offline and determined it's good to leave as is.

In short, we are willing to compress/not wrap functionality that is unlikely to be needed by developers to modify the plugin or build their own with the wrapped classes. Otherwise, we can expose functionality as closely as possible for clarity.